test/functional/lua/fs_spec.lua

```diff
@@ -291,9 +291,18 @@ describe('vim.fs', function()
     it('works with /', function()
       eq('/', vim.fs.normalize('/'))
     end)
+    it('works with / (opts.collapse_slash = false)', function()
+      eq('///', vim.fs.normalize('///', { collapse_slash = false }))
+    end)
     it('works with ~', function()
       eq(vim.fs.normalize(vim.uv.os_homedir()) .. '/src/foo', vim.fs.normalize('~/src/foo'))
     end)
+    it('removes duplicated slashes (opts.collapse_slash = true)', function()
+      eq('/home/user', vim.fs.normalize('//home//user//'))
+    end)
+    it('do not remove duplicated slashes (opts.collapse_slash = false)', function()
+      eq('//home//user', vim.fs.normalize('//home//user//', { collapse_slash = false })) -- does remove trailing slashes
```

Comment:
This makes no sense. If `normalize()` doesn't trim the last slashes, like this:

    //home/user/

then its result won't be the same for these inputs:

    //home//user//
    //home/user/

which defeats the purpose of normalizing. The normalized result should always be the same.

Reply:
It **does** remove any amount of trailing slashes, and keeps any duplicated slashes before the end of the path. Does that make sense?

Reply:
No. `//home//user//` and `//home/user/` are equivalent paths, yet `normalize({collapse_slash=false})` would return different results. Why can't it return `//home/user` if the input is `//home//user//`?

Reply:
As the literal option name suggests, this PR was intended to toggle whether to collapse multiple slashes into one. 

I didn't want to talk about any particular file system but more from a programming perspective and stop `gsub("/+", ...)` since it is an irreversible operation. 

I think fs.normalize has three missions. 

1. Change all `\` to `/`
2. Expand envvars (eg $HOME)
3. Shrink duplicated slashes. 

As we do have an option to toggle `2.`, I thought why couldn't there be an option to toggle `3.`?

Reply:
> I think fs.normalize has three missions.

No, its mission is to produce a canonical result for all equivalent forms of any kind of path.

Reply:
This is a nice idea. Added!

Reply:
@dundergoc Capitalize drive name.